### PR TITLE
Fix complex spread props breaking (#329)

### PR DESCRIPTION
**Pull request by pudility**
Fixing issue #329. 

TODO:

- [X] Add test cases for `this.props.foo`
- [X] Unlimited length `MemberExpression`s

### CODE REVIEW

File: src/_utils.js [hunk 1]
```diff
@@ -31,10 +31,18 @@ export const addClassName = (path, jsxId) => {
     const node = attr.node
     if (t.isJSXSpreadAttribute(attr)) {
       const name = node.argument.name
-      const attrNameDotClassName = t.memberExpression(
+      let attrNameDotClassName = t.memberExpression(
         t.identifier(name),
         t.identifier('className')
       )
+
+      if (node.argument.type === 'MemberExpression') {
```
Comment (giuseppeg) [inline src/_utils.js]: here you can do the following:

```js
let attrNameDotClassName

if (t.isMemberExpression(node.argument)) {
   attrNameDotClassName = // with nested member expression
} else {
   attrNameDotClassName = // with simple identifier
}
```

File: test/__snapshots__/index.js.snap [hunk 2]
```diff
@@ -81,6 +81,47 @@ export default class {
 }"
 `;
 
+exports[`works with complex spread props (class) 1`] = `
```
Comment (giuseppeg) [inline test/__snapshots__/index.js.snap]: Can you revert the changes to this file? It seems that you removed the fixture and the test from `test/index.js` but the snapshot is still here in the diff :)

Comment (pudility) [reply test/__snapshots__/index.js.snap]: I added back the tests in `test/index.js`, is that what you where looking for? If not I am not sure I understand what you are asking, sorry.

File: test/index.js [hunk 3]
```diff
@@ -20,6 +20,13 @@ test('works with stateless', async t => {
   t.snapshot(code)
 })
 
+test('works with spread', async t => {
```
Comment (giuseppeg) [inline test/index.js]: we don't need this test here since it is already in `test/attribute.js`

File: test/__snapshots__/index.js.snap (hunk 2 above)
Comment (giuseppeg) [reply test/__snapshots__/index.js.snap]: Sorry I didn't remember correctly the tests for this feature are in `test/attribute.js`. 
Can you do the following:

* remove the test from `test/index.js`
* regenerate the snapshot for `test/index.js` like so:

```
rm test/__snapshots__/index.js.snap
npm test -- -u
```

File: src/_utils.js [hunk 4]
```diff
@@ -31,10 +31,20 @@ export const addClassName = (path, jsxId) => {
...
+        attrNameDotClassName = t.memberExpression(
+          node.argument,
+          t.identifier('className')
+        )
+      } else {
+        attrNameDotClassName = t.memberExpression(
+          t.identifier(name),
+          t.identifier('className')
+        )
+      }
```
Comment (giuseppeg) [inline src/_utils.js]: nitpick: I just realized that this could be:

```js
const attrNameDotClassName = t.memberExpression(
  t.isMemberExpression(node.argument) ? node.argument : t.identifier(name),
  t.identifier('className')
)
```

**Comment by pudility** [comment]
I think I got it all, but if there is anything else I can change, just let me know :)

**Comment by giuseppeg** [comment]
@pudility awesome, thank you again for contributing!

**Comment by pudility** [comment]
@giuseppeg No problem, thanks for all the mentoring.
